homeassistant/components/zeroconf/usage.py

```diff
@@ -0,0 +1,14 @@
+"""Zeroconf usage utility to warn about multiple instances."""
+
+import zeroconf
+
+from homeassistant.helpers.frame import warn_use
+
+
+def install_multiple_zeroconf_warning() -> None:
+    """Wrap the Zeroconf class to warn if multiple instances are detected."""
+
+    zeroconf.Zeroconf.__init__ = warn_use(  # type: ignore
```

Comment:
This actually doesn't wrap it, it replaces the method. So this would prevent any Zeroconf from being created.

Reply:
The more I think about this, it might be better warn and then to hand them the shared instance in this case.

If we allow them to create another instance, we are heading for a cpu creep up (or a fast one in the spotcast case) until the user restarts or we get a hard crash.

Reply:
That is a great idea. And since we patch the close function, we don't have to be afraid that the code closes it either.